cmd/kubectl_token.go

```diff
@@ -510,58 +532,84 @@ func getAuthProviders(server string) (map[string]string, error) {
 		return nil, err
 	}
 
-	data := map[string]interface{}{}
-	err = json.Unmarshal(response, &data)
-	if err != nil {
-		return nil, err
+	if !gjson.ValidBytes(response) {
+		return nil, errors.New("invalid JSON input")
 	}
 
-	providers := map[string]string{}
-	i := 0
-	for _, value := range convert.ToMapSlice(data["data"]) {
-		provider := convert.ToString(value["type"])
-		if provider != "" && supportedAuthProviders[provider] {
-			providers[fmt.Sprintf("%v", i)] = provider
-			i++
+	data := gjson.GetBytes(response, "data").Array()
+
+	supportedProviders := []TypedProvider{}
```

Comment:
> the nil slice is the preferred style

https://go.dev/wiki/CodeReviewComments#declaring-empty-slices

I'll be honest, this surprises me, and I am guilty of not doing this :-)